tools/system_libs.py

```diff
@@ -600,12 +600,15 @@ def get_cflags(self):
     if self.includes:
       cflags += ['-I' + utils.path_from_root(i) for i in self._inherit_list('includes')]
 
+    source_dir = utils.path_from_root()
+    relative_source_dir = os.path.relpath(source_dir, self.build_dir)
     if self.deterministic_paths:
-      source_dir = utils.path_from_root()
-      relative_source_dir = os.path.relpath(source_dir, self.build_dir)
-      cflags += [f'-ffile-prefix-map={relative_source_dir}={DETERMINISITIC_PREFIX}']
       cflags += [f'-ffile-prefix-map={source_dir}={DETERMINISITIC_PREFIX}',
+                 f'-ffile-prefix-map={relative_source_dir}={DETERMINISITIC_PREFIX}',
                  f'-fdebug-compilation-dir={DETERMINISITIC_PREFIX}']
```

Comment:
`if deterministic_paths` part hasn't changed; I only merged two `cflags += ...` into one and changed the order of the flags.